### PR TITLE
Attach listeners to queue under a lock.

### PR DESCRIPTION
**Pull request by hamidp**
This fixes the following scenario:

Emitter service starts, attaches a listener to a queue with lots of items
Service gets notified of new items
Service emits an item and removes it from a queue while another thread goes in to add an item
Corruption!

Attaching a lock around setListener addresses this.

### CODE REVIEW

File: library/src/main/java/com/trello/snowman/EventQueue.java [hunk 1]
```diff
@@ -103,17 +103,19 @@ public void onRemove(ObjectQueue<EventOperation> queue) {
...
+                    }
+                    catch (FileException e) {
+                        SnowmanLog.e(TAG, "Error attaching listener, retrying.", e);
+
+                        clearAndDeleteQueueFile();
+                        createFileObjectQueue();
+
+                        // If it fails here it's not recoverable.
+                        mFileObjectQueue.setListener(listener);
+                    }
```
Comment (ulyssesp) [inline library/src/main/java/com/trello/snowman/EventQueue.java]: It feels like this block should be a  separate function outside of `Obsvable::create` that takes a listener parameter. `attachListenerToFileQueue()` or something maybe.

File: library/src/main/java/com/trello/snowman/EventQueue.java [hunk 2]
```diff
@@ -103,17 +103,19 @@ public void onRemove(ObjectQueue<EventOperation> queue) {
...
                 }
```
Comment (ulyssesp) [inline library/src/main/java/com/trello/snowman/EventQueue.java]: It feels like this block should be a separate function outside of Obsvable::create that takes a listener parameter. attachListenerToFileQueue() or something maybe.

Comment (hamidp) [inline library/src/main/java/com/trello/snowman/EventQueue.java]: Hm. What would the benefit be?

Comment (ulyssesp) [inline library/src/main/java/com/trello/snowman/EventQueue.java]: Readability. Decoupling setting a listener on `mFileObjectQueue` from `RxJava`. It's completely separate logic, right?

Comment (hamidp) [inline library/src/main/java/com/trello/snowman/EventQueue.java]: I had a variant where I extracted the `ObjectQueue.Listener` into a separate class but it wasn't that much better. Extracting a method didn't really help much either.

**Comment by hamidp** [comment]
This actually doesn't help, closing.

**Pull request closed without merging**